### PR TITLE
I hate git

### PR DESCRIPTION
**Pull request by Groders**
No idea if this will work, but gl gl

### CODE REVIEW

File: files/mkafka.c [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+static mkafka_t *mkafka;
+static int mutex;
+
+
+/*
+ * Used for debugging purposes
+ * Print all messages which are currently in the mailbox
+
+int print_all_messages()
```
Comment (kwojcicki) [inline files/mkafka.c]: commented because not used or because doesnt work?

File: files/mkafka.c [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    }
+
+    return 0;
+ }
+*/
+
+/* initialized mkafka  */
+int init_mkafka (){
+    mkafka = malloc (sizeof(mkafka_t));
+    mkafka->head = NULL;;
```
Comment (kwojcicki) [inline files/mkafka.c]: ;;

**Comment by kwojcicki** [comment]
@Groders  can you include the test files you did?

File: files/mkafka.c (hunk 1 above)
Comment (Groders) [reply files/mkafka.c]: both :), we should eventually make it work, as well as add a system call to call it.

File: files/mkafka.c (hunk 2 above)
Comment (Groders) [reply files/mkafka.c]: style choice, fix it your self ;)

File: files/mkafka.c [hunk 3]
```diff
@@ -0,0 +1,293 @@
+/* File mailbox.c */
+/* Define system calls to the mailbox */
+
+#include "mkafkalib.h"
+#include <minix/syslib.h>
+#include "glo.h"
+
+
+static mkafka_t *mkafka;
+static int mutex;
```
Comment (kwojcicki) [inline files/mkafka.c]: i just realized your "mutex" is an int....

File: files/mkafka.c [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+
+/* Creates mailbox if it is missing
+ * Add message to mailbox (if mailbox is not full)
+ * Returns OK if message was successfully added
+ * Returns ERROR if mailbox is full
+ */
+int push_to_topic()
+{
+    while (mutex==1);
```
Comment (kwojcicki) [inline files/mkafka.c]: bless the cpu

Comment (Groders) [reply files/mkafka.c]: I don't actually know how to make this not shit, i'm putting you in charge :).

File: files/mkafka.c (hunk 3 above)
Comment (Groders) [reply files/mkafka.c]: :)

**Pull request merged**